### PR TITLE
[15.0][ADD] event_stage_cancelled: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Up to v14, events had an state field instead of configurable stages. A lost feature with that change was the concept of a cancelled event and with it the logic associated to it: when we cancelled an event, its registrations were cancelled along with it.

@Tecnativa TT47686

please review @pedrobaeza

### CODE REVIEW

File: event_stage_cancelled/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2020 Tecnativa - David Vidal
```
Comment (pedrobaeza) [inline event_stage_cancelled/__manifest__.py]: Don't live in the past :stuck_out_tongue_closed_eyes: 

```suggestion
# Copyright 2024 Tecnativa - David Vidal
```

File: event_stage_cancelled/readme/DESCRIPTION.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[ This file must be max 2-3 paragraphs, and is required. ]
```
Comment (pedrobaeza) [inline event_stage_cancelled/readme/DESCRIPTION.md]: Missing file.

**Review by pedrobaeza — COMMENTED** [review]
Please put the flag `is_cancelled` in `event.event_stage_cancelled` as data of the module.

**Comment by chienandalu** [comment]
All done :smile:

File: event_stage_cancelled/readme/DESCRIPTION.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+This module adds the posibility of flaggin an event stage as cancelled so we can hook
```
Comment (pedrobaeza) [inline event_stage_cancelled/readme/DESCRIPTION.md]: I would put here all the speech about that this was present in v13 and so on, and also mention that the cancellation leads to registrations cancelling as well.

Comment (chienandalu) [reply event_stage_cancelled/readme/DESCRIPTION.md]: Doesn't it fit better in the *Context* section?

Comment (pedrobaeza) [reply event_stage_cancelled/readme/DESCRIPTION.md]: Yes, I didn't remember that new section.

**Comment by pedrobaeza** [comment]
Please include in the README the new features.

**Comment by pedrobaeza** [comment]
On runboat, I have cancelled an event (http://oca-event-15-0-pr363-73192864b7ce.runboat.odoo-community.org/web#id=8&cids=1&menu_id=134&action=204&model=event.event&view_type=form), defining a mail action, but no mail was sent.

And another thing: we need a cancel mail template.

**Comment by chienandalu** [comment]
> On runboat, I have cancelled an event (http://oca-event-15-0-pr363-73192864b7ce.runboat.odoo-community.org/web#id=8&cids=1&menu_id=134&action=204&model=event.event&view_type=form), defining a mail action, but no mail was sent.

Did you run the cron? It's the one in charge of handling the schedulers. And after the scheduler is settled you should force the outgoing mail queue.

Anyway, I just improved the reliability as I was seeing when doing the test cases some pitfalls.

> And another thing: we need a cancel mail template.

:+1:

**Comment by chienandalu** [comment]
Ready to test

**Comment by pedrobaeza** [comment]
I don't get the mails to work. I have run manually "Event Email Reminder" cron also. The event is http://oca-event-15-0-pr363-5af3d44b1b5e.runboat.odoo-community.org/web#id=8&cids=1&menu_id=134&action=204&model=event.event&view_type=form

**Comment by chienandalu** [comment]
> I don't get the mails to work...

Ok, it's just an incompatibility issue with `event_registration_cancel_reason`. I just added a flagged context to avoid it. A compatibility module might be needed but we don't have any customer making use of these two at once.

**Review by pedrobaeza — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Pull request merged**